### PR TITLE
rpc,server: use node certs for shared-process tenant RPCs

### LINKED ISSUE #96215: server: make shared-process tenant servers use node certs for outgoing RPC connections
**Describe the problem**

Currently, *outgoing* RPC connections across the cluster by secondary tenant servers still require using the special "tenant client cert", to be manually minted before the tenant is created.

Since the tenant is running in the same process as the KV layer, there is no good reason for it to needs a separate cert. Having per-tenant certs in that case also makes orchestration more difficult.

**Expected behavior**

Outgoing RPCs from secondary tenant servers running in the same process as KV should use the same certs as KV does itself.

The solution for this requires first fixing #75795.

Epic: CRDB-23559

Jira issue: CRDB-24008

### PR DESCRIPTION
**Pull request by knz**
TLDR: This PR makes it possible to use regular node certs with shared-process multi-tenancy (i.e. skip creating client tenant certs). I think this will help @aadityasondhi 's work on `debug zip`.

Fixes #96215.
Epic: CRDB-23559

The PR also contains a few cleanup commits; it's a short change away from fixing  #87141 and #77173, which I intend to do in a separate PR.

### CODE REVIEW

**Review by stevendanna — APPROVED** [review]
Thanks for pushing this forward.

**Comment by knz** [comment]
thank you!

bors r=stevendanna

**Comment by knz** [comment]
bors r=stevendanna
